apps/supernova/docker/Dockerfile

```diff
@@ -73,56 +26,25 @@ ENV SILENCE_EXCLUDED_LABELS=""
 ENV SILENCE_TEMPLATES=""
 ENV PREDEFINED_FILTERS=""
 
-# Create Entrypoint which creates /app/appProps.js from ENVs
-# Environment variables are known at runtime, so we can use them to create a file that will be used by the app
-RUN cat <<EOF > /supernova_entrypoint.sh
-#!/bin/sh
-# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
-# SPDX-License-Identifier: Apache-2.0
-
-# use jq to overwrite the default values in appProps.json
-# if the corresponding environment variables are set
-
-# Define the JSON file path
-APP_PROPS_FILE="/usr/share/nginx/html/appProps.json"
-
-# Copy the appProps.json file to the /usr/share/nginx/html directory if it exists
-if [ -f /appProps.json ]; then
-cp /appProps.json "\${APP_PROPS_FILE}"
-fi
-
-# Function to update JSON fields using jq
-update_json_field() {
-  field="\$1"
-  value="\$2"
-  if [ -n "\$value" ]; then
-    # Ensure jq and temporary file handling are POSIX compliant
-    jq --arg v "\$value" ".\${field} = \\\$v" "\$APP_PROPS_FILE" > tmp.json && mv tmp.json "\$APP_PROPS_FILE"
-  fi
+# default appProps.json
+RUN cat <<EOF > /appProps.json
+{
+  "endpoint": "",
+  "theme": "theme-dark",
+  "embedded": false,
+  "silenceExcludedLabels": "",
+  "silenceTemplates": ""
 }
+EOF
 
-# Update fields if corresponding environment variables are set
-update_json_field "endpoint" "\$ENDPOINT"
-update_json_field "theme" "\$THEME"
-update_json_field "embedded" "\$EMBEDDED"
-update_json_field "silenceExcludedLabels" "\$SILENCE_EXCLUDED_LABELS"
-update_json_field "silenceTemplates" "\$SILENCE_TEMPLATES"
-update_json_field "predefinedFilters" "\$PREDEFINED_FILTERS"
-update_json_field "initialFilters" "\$INITIAL_FILTERS"
-update_json_field "username" "\$USERNAME"
-
-echo "\$APP_PROPS_FILE has been updated."
-
-cat "\$APP_PROPS_FILE"
+COPY --from=build /app/apps/supernova/build /usr/share/nginx/html/
 
-exec "\$@"
-EOF
+COPY apps/greenhouse/docker/nginx.conf /etc/nginx/conf.d/default.conf
+COPY apps/greenhouse/docker/entrypoint.sh /usr/local/bin/entrypoint.sh 
 
-RUN chmod +x /supernova_entrypoint.sh
+RUN chmod +x /usr/local/bin/entrypoint.sh
```

Comment:
this step could be removed when we change the file permissions before checking in. Just do `chmod +x /usr/local/bin/entrypoint.sh` and push the changes.

On the other side you need the `entrypoint.sh` as executable to run it for test reasons on your dev system so it is always a good idea to made it executable by default :-)

Reply:
Good catch! Let’s make sure to add it to the list of improvements.